sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -658,6 +658,18 @@ def test_radius_neighbors_regressor(n_samples=40,
             y_pred = neigh.predict(X[:n_test_pts] + epsilon)
             assert_true(np.all(abs(y_pred - y_target) < radius / 2))
 
+            # test fix for issue #9654
+            # test that nan is returned when no nearby observations
+            try:
+                y_pred_alt = neigh.predict([[-1]])
+                y_pred_alt_isnan = np.all(np.isnan(y_pred_alt))
+                raise_zero_div_error = False
+
+            except ZeroDivisionError:
```

Comment:
This is unnecessary. The test will fail if there is an error raised.

I was asking about handling the warning which numpy issues when performing `np.mean([])`. I'm not sure whether it's good or bad for this warning to be issued, but we should be consistent in the weighted case. warnings are tested with our `assert_warns_message` and `assert_no_warnings` helpers.

Reply:
Right, good point. 

I looked into raising the warning - apparently warnings only arise when there are uniform weights. What  do you think of the following solution?

```python
# test fix for issue #9654
# test that nan is returned when no nearby observations        
X_test_nan = np.ones((1,n_features))*-1
if weights=='uniform':        
    assert_warns_message(RuntimeWarning, "Mean of empty slice.", 
                         neigh.predict, X_test_nan)

assert_true(np.all(np.isnan(neigh.predict(X_test_nan))))
```